wordpress/plugins/kredeum-nfts/admin/ajax/ajax.php

```diff
@@ -28,10 +28,13 @@ function () {
 	function () {
 		check_ajax_referer( 'ajax-token', 'security' );
 
-		if ( isset( $_POST['pid'] ) && isset( $_POST['nid'] ) ) {
+		if ( isset( $_POST['pid'] ) && isset( $_POST['nid'] ) && isset( $_POST['cid'] ) ) {
 			$pid = sanitize_text_field( wp_unslash( $_POST['pid'] ) );
 			$nid = sanitize_text_field( wp_unslash( $_POST['nid'] ) );
-			 add_post_meta( $pid, '_kre_nid', $nid, true );
+			$cid = sanitize_text_field( wp_unslash( $_POST['cid'] ) );
+			add_post_meta( $pid, '_kre_nid', $nid, true );
```

Comment:
why add_post_meta for nid and update_post_meta for cid ?

Reply:
in case of the user already posted CID by using bulk actions

Reply:
normaly not because the goal is to mint only once.
But yes we could use update_post_meta

Reply:
WP docs : "[update_post_meta()](https://developer.wordpress.org/reference/functions/update_post_meta/) Can be used in place of [add_post_meta()](https://developer.wordpress.org/reference/functions/add_post_meta/) ."

so replace it, yes